### PR TITLE
[WIP] Proposal: core concepts

### PR DESCRIPTION
**Pull request by pedronasser**

### CODE REVIEW

File: docs/concepts.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+**Function** is the unit responsible for handling a specific part of your service. Is usually formed by a small code that receives a payload, does its work and can return the result.
+
+You can [configure a function](#Link-to-the-function-file-doc) adding a file called `func.yml` in its source directory and setup how that function needs to run.
+
+A function will run a task inside a container when triggered by a [route](#Route).
+
+### Function Structure
+
+```
+image          string                   (username/function_name)
```
Comment (ccirello) [inline docs/concepts.md]: I guess this is `name` right?

File: docs/concepts.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+You can easily deploy multiple functions inside the same `App` creating a `app.yml` file in the parent directory of your functions directory.
+
+### App.yml Structure
+
+```
+name: name-of-the-app 
+
+functions:
+  - "/name-of-the-app/myRoute:{path to the function directory}"
```
Comment (ccirello) [inline docs/concepts.md]: Why not use regular maps in YAML?

Comment (pedronasser) [inline docs/concepts.md]: Good point.

File: docs/concepts.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+### Function Structure
+
+```
+name            string                   (username/function_name)
+memory          number                   (max amount of memory used by the function)
+headers         object{key => array}     (returned HTTP headers)
+type            string (sync/async)      (how the function will be running)
+format          string (http/-)          (how the function will receive the payload)
+max_concurrency number                   (max concurrent containers can we run)
```
Comment (ccirello) [inline docs/concepts.md]: What you say of making it clear that is per function per node?

File: docs/concepts.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+### App.yaml Structure
+
+```
+routes:
+  /name-of-the-app/myRoute: {path to the function directory}
+  /name-of-the-app/myOtherRoute:
+    function: myOtherRoute/
+    tag:
+      private: true
```
Comment (ccirello) [inline docs/concepts.md]: What is `private`?

Comment (pedronasser) [inline docs/concepts.md]: Just a tag example.

File: docs/concepts.md (hunk 3 above)
Comment (pedronasser) [inline docs/concepts.md]: Sure. Updating

File: docs/concepts.md (hunk 4 above)
Comment (ccirello) [inline docs/concepts.md]: Oh... could you please then use a more example-like name? Because it felt it was some sort of IronFunctions attribute for me.

Comment (pedronasser) [inline docs/concepts.md]: Oh okay. Good point

**Comment by pedronasser** [comment]
The idea of this PR is define the future.

**Comment by pedronasser** [comment]
Deferred

**Pull request closed without merging**